### PR TITLE
mds: add command that config individual client session

### PR DESCRIPTION
**Pull request by ukernel**
For now, the command only can change session's timeout config

Fixes: http://tracker.ceph.com/issues/40811
Signed-off-by: "Yan, Zheng" <zyan@redhat.com>

- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

### CODE REVIEW

File: src/mds/MDSDaemon.cc [hunk 1]
```diff
@@ -563,6 +570,11 @@ const std::vector<MDSDaemon::MDSCommand>& MDSDaemon::get_commands()
     MDSCommand("client ls name=filters,type=CephString,n=N,req=false", "List client sessions"),
     MDSCommand("session evict name=filters,type=CephString,n=N,req=false", "Evict client session(s)"),
     MDSCommand("client evict name=filters,type=CephString,n=N,req=false", "Evict client session(s)"),
+    MDSCommand("session config name=client_id,type=CephInt name=option,type=CephString name=value,type=CephString,req=false",
+	"Config a client session"),
+    MDSCommand("client config name=client_id,type=CephInt name=option,type=CephString name=value,type=CephString,req=false",
+	"Config a client session"),
+    MDSCommand("damage ls", "List detected metadata damage"),
```
Comment (batrick) [inline src/mds/MDSDaemon.cc]: looks like you have some leftover copy/paste

File: src/mds/MDSDaemon.cc [hunk 2]
```diff
@@ -271,16 +271,23 @@ void MDSDaemon::set_up_admin_socket()
...
+				     "session config name=client_id,type=CephInt,req=true "
+				     "name=option,type=CephString,req=true "
+				     "name=value,type=CephString,req=false ",
+				     asok_hook,
+				     "Config a CephFS client session");
+  assert(r == 0);
+  r = admin_socket->register_command("osdmap barrier",
+				     "osdmap barrier name=target_epoch,type=CephInt",
+				     asok_hook,
+				     "Wait until the MDS has this OSD map epoch");
```
Comment (batrick) [inline src/mds/MDSDaemon.cc]: why move this code?

File: src/mds/MDSRank.cc [hunk 3]
```diff
@@ -2489,13 +2489,20 @@ bool MDSRankDispatcher::handle_asok_command(std::string_view command,
     }
 
     mds_lock.Lock();
-    std::stringstream dss;
-    bool evicted = evict_client(strtol(client_id.c_str(), 0, 10), true,
-        g_conf()->mds_session_blacklist_on_evict, dss);
-    if (!evicted) {
-      dout(15) << dss.str() << dendl;
-      ss << dss.str();
-    }
```
Comment (batrick) [inline src/mds/MDSRank.cc]: why remove this debug message?

File: src/mds/MDSDaemon.cc (hunk 2 above)
Comment (ukernel) [reply src/mds/MDSDaemon.cc]: put 'session foo' together

File: qa/tasks/cephfs/test_client_recovery.py [hunk 4]
```diff
@@ -601,3 +601,21 @@ def test_dont_mark_unresponsive_client_stale(self):
...
+
+    def test_config_session_timeout(self):
+        self.fs.mds_asok(['config', 'set', 'mds_defer_session_stale', 'false'])
+        session_timeout = self.fs.get_var("session_timeout")
+        mount_a_gid = self.mount_a.get_global_id()
+
+        self.fs.mds_asok(['session', 'config', '%s' % mount_a_gid, 'timeout', '%s' % (session_timeout * 2)])
+
+        self.mount_a.kill();
+        self.mount_a.kill_cleanup()
```
Comment (batrick) [inline qa/tasks/cephfs/test_client_recovery.py]: `kill_cleanup` now waits for the hard reboot to complete: #28825 

I guess the right thing to do here is to add a sleep after `kill()`, assert 2 sessions, then do `kill_cleanup()`.

See this failure:

/ceph/teuthology-archive/pdonnell-2019-07-26_06:38:30-kcephfs-wip-pdonnell-testing-20190726.021409-distro-basic-smithi/4152101/teuthology.log

**Pull request merged**